libraries/chain/apply_context.cpp

```diff
@@ -61,51 +61,53 @@ void apply_context::exec_one()
       try {
          action_return_value.clear();
          receiver_account = &db.get<account_metadata_object,by_name>( receiver );
-         privileged = receiver_account->is_privileged();
-         auto native = control.find_apply_handler( receiver, act->account, act->name );
-         if( native ) {
-            if( trx_context.enforce_whiteblacklist && control.is_producing_block() ) {
-               control.check_contract_list( receiver );
-               control.check_action_list( act->account, act->name );
+         if( !context_free || (context_free && !control.skip_trx_checks() ) ) {
```

Comment:
The second `context_free` is redundant.

Maybe for clarity you can rewrite as:
```cpp
if( !(context_free && control.skip_trx_checks()) ) {
```

Reply:
done